pegen/pegen.c

```diff
@@ -821,20 +840,32 @@ construct_assign_target(Parser *p, expr_ty node)
         case Name_kind:
             return _Py_Name(node->v.Name.id,
                             Store,
-                            EXTRA_EXPR(node, node));
+                            node->lineno,
+                            node->col_offset,
+                            node->end_lineno,
+                            node->end_col_offset,
+                            p->arena);
         case Tuple_kind:
             if (asdl_seq_LEN(node->v.Tuple.elts) != 1) {
                 PyErr_Format(PyExc_SyntaxError, "Only single target (not tuple) can be annotated");
                 //TODO: We need to return a dummy here because we don't have a way to correctly
                 // buble up exceptions for now.
                return _Py_Name(_create_dummy_identifier(p),
                             Store,
-                            EXTRA_EXPR(node, node));
+                            node->lineno,
+                            node->col_offset,
+                            node->end_lineno,
+                            node->end_col_offset,
+                            p->arena);
```

Comment:
Seems maybe you could keep `EXTRA_EXPR` around just so it can be called by various helpers in pegen.c?